### PR TITLE
add unit test to pagination under routes.

### PR DESCRIPTION
**Pull request by mgold1234**
the percent of unit test was very low 20%
this pr increase the covarage of unit test

Signed-off-by: mgold1234 <mgold@redhat.com>

# Description

FIXES: 

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

File: pkg/routes/common/pagination_test.go [hunk 1]
```diff
@@ -46,3 +49,52 @@ func TestGetPagination(t *testing.T) {
...
+	for _, te := range tt {
+		t.Run(te.name, func(t *testing.T) {
+			router := chi.NewRouter()
+			router.Use(Paginate)
+			router.Get("/", func(w http.ResponseWriter, r *http.Request) {
+				pagination, ok := r.Context().Value(PaginationKey).(Pagination)
+				Expect(ok).To(BeTrue())
+				Expect(pagination.Limit).To(Equal(te.expected.Limit))
+				Expect(pagination.Offset).To(Equal(te.expected.Offset))
+			})
```
Comment (ldjebran) [inline pkg/routes/common/pagination_test.go:75]:
```suggestion
                             w.WriteHeader(http.StatusOK)
			})
```

File: pkg/routes/common/pagination_test.go [hunk 2]
```diff
@@ -46,3 +49,52 @@ func TestGetPagination(t *testing.T) {
...
+			path := ""
+			if te.limit > 0 {
+				path = fmt.Sprintf("limit=%d", te.limit)
+			}
+			if te.offset > 0 {
+				offset := fmt.Sprintf("offset=%d", te.offset)
+				if path == "" {
+					path = offset
+				} else {
+					path = path + " " + offset
```
Comment (ldjebran) [inline pkg/routes/common/pagination_test.go:85]: very strange that this is working 
```suggestion
					path = path + "&" + offset
```

File: pkg/routes/common/pagination_test.go [hunk 3]
```diff
@@ -46,3 +49,52 @@ func TestGetPagination(t *testing.T) {
...
+		limit    int
+		offset   int
+		expected Pagination
+	}{
+		{name: "check limit value 10 offset 20", limit: 10, offset: 20, expected: Pagination{Limit: 10, Offset: 20}},
+		{name: "check limit value 10 offset 0", limit: 10, offset: 0, expected: Pagination{Limit: 10, Offset: defaultOffset}},
+		{name: "check limit value 0 offset 0", limit: 0, offset: 0, expected: Pagination{Limit: defaultLimit, Offset: defaultOffset}},
+		{name: "check limit value 10 offset 20", limit: 0, offset: 20, expected: Pagination{Limit: defaultLimit, Offset: 20}},
+	}
+
```
Comment (ldjebran) [inline pkg/routes/common/pagination_test.go:65]: you can create a simple function getPath
```go
       getPath := func(limit, offset int) string {
		path := "/"
		params := make([]string, 0, 2)
		if limit > 0 {
			params = append(params, fmt.Sprintf("limit=%d", limit))
		}
		if offset > 0 {
			params = append(params, fmt.Sprintf("offset=%d", offset))
		}
		if len(params) > 0 {
			path = path + "?" + strings.Join(params, "&")
		}
		return path
	}
```

and use it later in the subtests , that became more compact

```go
req, err := http.NewRequest("GET", getPath(te.limit, te.offset), nil)
```

**Review by ldjebran — DISMISSED** [review]
/lgtm

**Review by ldjebran — DISMISSED** [review]

File: pkg/routes/common/pagination_test.go [hunk 4]
```diff
@@ -46,3 +50,52 @@ func TestGetPagination(t *testing.T) {
...
+	if offset > 0 {
+		params = append(params, fmt.Sprintf("offset=%d", offset))
+	}
+
+	if len(params) > 0 {
+		path = path + "?" + strings.Join(params, " ")
+	}
+
+	return path
+}
```
Comment (omaciel) [inline pkg/routes/common/pagination_test.go:68]: Add a new line

File: pkg/routes/common/pagination_test.go [hunk 5]
```diff
@@ -46,3 +50,52 @@ func TestGetPagination(t *testing.T) {
 	}
 
 }
+func getPath(limit, offset int) string {
```
Comment (omaciel) [inline pkg/routes/common/pagination_test.go:53]: Add new line above

File: pkg/routes/common/pagination_test.go [hunk 6]
```diff
@@ -46,3 +50,52 @@ func TestGetPagination(t *testing.T) {
...
+		{name: "check limit value 0 offset 0", limit: 0, offset: 0, expected: Pagination{Limit: defaultLimit, Offset: defaultOffset}},
+		{name: "check limit value 10 offset 20", limit: 0, offset: 20, expected: Pagination{Limit: defaultLimit, Offset: 20}},
+	}
+
+	for _, te := range tt {
+		t.Run(te.name, func(t *testing.T) {
+			router := chi.NewRouter()
+			router.Use(Paginate)
+			router.Get("/", func(w http.ResponseWriter, r *http.Request) {
+				pagination, ok := r.Context().Value(PaginationKey).(Pagination)
```
Comment (omaciel) [inline pkg/routes/common/pagination_test.go:88]: Should the last argument be te.Pagination?

**Review by omaciel — COMMENTED** [review]
Quick comments and a question?

File: pkg/routes/common/pagination_test.go (hunk 6 above)
Comment (mgold1234) [reply pkg/routes/common/pagination_test.go:88]: no, the 'te' has those fields: 
name,limit, offset, expected
and we compare pagination.limit from the context te. expected.pagination
hope it is understandable, if not tell me and I will explain with more details

Comment (ldjebran) [reply pkg/routes/common/pagination_test.go:90]: that Pagination is a type , we need to cast the context value to Pagination type

```go
type Pagination struct {
...
```

**Review by ldjebran — APPROVED** [review]

File: pkg/routes/common/pagination_test.go (hunk 6 above)
Comment (omaciel) [reply pkg/routes/common/pagination_test.go:90]: @ldjebran does this need to be resolved then?

**Review by omaciel — COMMENTED** [review]
Checking if this comment can be resolved?

File: pkg/routes/common/pagination_test.go (hunk 6 above)
Comment (ldjebran) [reply pkg/routes/common/pagination_test.go:90]: @omaciel yes we need to be resolved that's why we should have 
```go
Expect(ok).To(BeTrue())
```

Comment (omaciel) [reply pkg/routes/common/pagination_test.go:90]: @ldjebran confirmed he's good with the code. Marking as resolved.

**Review by omaciel — APPROVED** [review]
/lgtm

**Pull request merged**